### ISSUE TITLE
fiasco: build errors with gcc 12

### ISSUE THREAD

**Issue by cproc** [genodelabs/genode#4836]
When building the Fiasco platform with gcc 12, errors like the following occur:

```
      [sigma0/server]  /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/src/kernel/fiasco/fiasco/snapshot/l4/pkg/sigma0/server/src/init.cc:31: undefined reference to `strlen'
```

**Comment by cproc** [comment]
Fixed by 9b4bfa5.

**Comment by cproc** [comment]
Another error:

```
      [sigma0/server]  In file included from /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/sr
c/kernel/fiasco/fiasco/snapshot/l4/pkg/sigma0/server/src/mem_man.h:7,
      [sigma0/server]                   from /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/sr
c/kernel/fiasco/fiasco/snapshot/l4/pkg/sigma0/server/src/mem_man.cc:1:
      [sigma0/server]  /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/src/kernel/fiasco/fiasco
/snapshot/l4/pkg/sigma0/server/src/region.h: In instantiation of 'OS& operator<<(OS&, const Region&) [with OS = L4::BasicOStream]':
      [sigma0/server]  /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/src/kernel/fiasco/fiasco
/snapshot/l4/pkg/sigma0/server/src/mem_man.cc:213:26:   required from here
      [sigma0/server]  /.../contrib/fiasco-386db79cbd4039ea2e3cbf028fac095a1bc96c31/src/kernel/fiasco/fiasco/snapshot/l4/pkg/sigma0/server/src/region.h:49:6: error: no match for 'operator<<' (operand types are 'L4::BasicOStream' and 'char')
      [sigma0/server]     49 |   os << '[' << r.owner() << ':' << (void*)r.start() << ';' << (void*)r.end() << ']';
      [sigma0/server]        |   ~~~^~~~~~
```

Fixed by 0227d91.
